tests/run/test_Saver.py

```diff
@@ -0,0 +1,113 @@
+        ]
+        seq_to_seq.lr_scheduler._step_count += 2
+
+
+# Tests -----------------------------------------------------------------------
+
+
+@settings(deadline=3000)
+@given(seq_to_seq_cfg=task_configs(), check_with_fit_api=st.booleans())
+def test_seq_to_seq_correctly_built_saved_and_loaded(
+    seq_to_seq_cfg: task_config_pb2.TaskConfig, check_with_fit_api: bool,
+) -> None:
+    """Test that SeqToSeq is correctly built + saved + loaded."""
+    seq_to_seq, epochs, train_loader, _ = build_and_check_task_config(
+        seq_to_seq_cfg
+    )
+
+    with TemporaryDirectory() as tmpdir:
+        # Init Saver Callback, and simulate x2 epochs to save state_dict
+        cb_handler = CallbackHandler(
+            callbacks=[Saver(log_dir=tmpdir, model=seq_to_seq)]
+        )
+
+        cb_handler.on_train_begin(epochs)
+        cb_handler.on_epoch_end()
+        cb_handler.on_epoch_end()  # saves state_dict
+
+        # make copy of saved state_dict
+        expected_sd = copy.deepcopy(seq_to_seq.state_dict())
+
+        # Alter state_dict so that it != expected_sd
+        change_seq_to_seq_state_dict(seq_to_seq)
+
+        # check that state_dict **has** changed by checking that all subdicts
+        # are altered
+        with pytest.raises(AssertionError):
+            check_state_dicts_match(
+                expected_sd["model"], seq_to_seq.state_dict()["model"]
+            )
+        if seq_to_seq.optim:
+            with pytest.raises(AssertionError):
+                check_state_dicts_match(
+                    expected_sd["optim"], seq_to_seq.state_dict()["optim"]
+                )
+        if seq_to_seq.lr_scheduler:
+            with pytest.raises(AssertionError):
+                check_state_dicts_match(
+                    expected_sd["lr_scheduler"],
+                    seq_to_seq.state_dict()["lr_scheduler"],
+                )
+
+        if not check_with_fit_api:
+            # Use **newly initialised** Saver cb to reload the state_dict
+            cb_handler = CallbackHandler(
+                callbacks=[Saver(log_dir=tmpdir, model=seq_to_seq)]
+            )
+
+            cb_handler.on_train_begin(epochs)
+
+            check_state_dicts_match(expected_sd, seq_to_seq.state_dict())
+            assert cb_handler.state_dict["epoch"] == 2
+            assert cb_handler.state_dict["total_train_batches"] == 0
+        else:
+            # It should also be possible to reload state_dict when fit
+            # orchestrates the CallbackHandler.
+
+            # 'Run training' for no epochs - the Saver callback should re-load
+            # seq_to_seq state_dict == expected_sd
+            fit(
+                seq_to_seq,
+                epochs=0,
+                train_loader=train_loader,
+                eval_loader=None,
+                callbacks=[Saver(log_dir=tmpdir, model=seq_to_seq)],
+            )
+
+            check_state_dicts_match(expected_sd, seq_to_seq.state_dict())
```

Comment:
Are both of these required? `fit` does the `not check_with_fit_api` branch internally?

Reply:
Sure thing- I thought it was useful to check that both work but it is probably overkill! I've removed the first branch.